demos/speech_recognition_demo/python/ctcdecode-numpy/ctcdecode_numpy/__init__.py

```diff
@@ -59,18 +128,18 @@ def decode(self, probs, seq_lens=None):
         return output, scores, timesteps, out_seq_len
 
     def character_based(self):
-        return ctc_decode.is_character_based(self._scorer) if self._scorer else None
+        return self._scorer.is_character_based() if self._scorer else None
 
     def max_order(self):
-        return ctc_decode.get_max_order(self._scorer) if self._scorer else None
+        return self._scorer.get_max_order() if self._scorer else None
 
     def dict_size(self):
-        return ctc_decode.get_dict_size(self._scorer) if self._scorer else None
+        return self._scorer.get_dict_size() if self._scorer else None
 
     def reset_params(self, alpha, beta):
         if self._scorer is not None:
-            ctc_decode.reset_params(self._scorer, alpha, beta)
+            self._scorer.reset_params(alpha, beta)
 
-    def __del__(self):
-        if self._scorer is not None:
-            ctc_decode.delete_scorer(self._scorer)
+
+# For compatibility with ctcdecode_numpy version 0.2.0
+CTCBeamDecoder = BatchedCtcLmDecoder
```

Comment:
We don't have a requirement to be backward compatible. You broke compatibility any way: at least with `labels`->`alphabet` renaming (I like the new name more :) ) I would completely remove `BatchedCtcLmDecoder` because it becomes dead since we won't ever run it.

I have perception that some things from underlying layers of implementation should also be removed following that principle, though I haven't figured out what happens here yet.
CC @vladimir-dudnik

Reply:
agree, no reasons to keep a dead code

Reply:
1. Removed `CTCBeamDecoder` alias for `BatchedCtcLmDecoder`.
2. `BatchedCtcLmDecoder` is used in Accuracy Checker, so the code is not dead.
3. The underlying C++ implementation is a batching wrapper (based on ThreadPool) over the same C++ class `CtcDecoderState` (which is single-threaded).